extensions/resteasy-observability/deployment/src/main/java/io/quarkus/resteasy/observability/deployment/RestPathAnnotationProcessor.java

```diff
@@ -0,0 +1,118 @@
+package io.quarkus.resteasy.observability.deployment;
+
+import java.util.Optional;
+import java.util.regex.Pattern;
+
+import org.jboss.jandex.AnnotationInstance;
+import org.jboss.jandex.AnnotationTarget;
+import org.jboss.jandex.AnnotationValue;
+import org.jboss.jandex.ClassInfo;
+import org.jboss.jandex.DotName;
+import org.jboss.jandex.MethodInfo;
+
+import io.quarkus.arc.deployment.AdditionalBeanBuildItem;
+import io.quarkus.arc.deployment.AnnotationsTransformerBuildItem;
+import io.quarkus.arc.processor.AnnotationsTransformer;
+import io.quarkus.deployment.Capabilities;
+import io.quarkus.deployment.Capability;
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.metrics.MetricsCapabilityBuildItem;
+import io.quarkus.resteasy.observability.runtime.QuarkusRestPathTemplate;
+import io.quarkus.resteasy.observability.runtime.QuarkusRestPathTemplateInterceptor;
+import io.quarkus.runtime.metrics.MetricsFactory;
+
+public class RestPathAnnotationProcessor {
+
+    static final DotName REST_PATH = DotName.createSimple("javax.ws.rs.Path");
+    static final DotName REGISTER_REST_CLIENT = DotName
+            .createSimple("org.eclipse.microprofile.rest.client.inject.RegisterRestClient");
+    static final DotName TEMPLATE_PATH = DotName.createSimple(QuarkusRestPathTemplate.class.getName());
+    static final DotName TEMPLATE_PATH_INTERCEPTOR = DotName.createSimple(QuarkusRestPathTemplateInterceptor.class.getName());
+
+    public static final Pattern MULTIPLE_SLASH_PATTERN = Pattern.compile("//+");
+
+    @BuildStep()
+    AdditionalBeanBuildItem registerBeanClasses(Capabilities capabilities,
+            Optional<MetricsCapabilityBuildItem> metricsCapability) {
+        AdditionalBeanBuildItem.Builder builder = AdditionalBeanBuildItem.builder()
+                .addBeanClass(TEMPLATE_PATH.toString())
+                .addBeanClass(TEMPLATE_PATH_INTERCEPTOR.toString());
+
+        if (!(metricsCapability.isPresent() && metricsCapability.get().metricsSupported(MetricsFactory.MICROMETER))
+                && capabilities.isMissing(Capability.OPENTELEMETRY_TRACER)) {
+            builder.setRemovable();
+        }
+
+        return builder.build();
```

Comment:
Rather than fuss with removable or not, just return null. Don't even create the beans.

Also, consider passing metricsCapability and capabilities into a readable method ( "hasConsumer" or something) that can be used in findRestPaths as well.